### PR TITLE
String input pane

### PR DESCRIPTION
**Pull request by hardiknai-techm**
https://user-images.githubusercontent.com/80053762/114806396-529a6b00-9dc2-11eb-851a-09644415009a.mp4

### CODE REVIEW

File: src/Explorer/Panes/StringInputPane/StringInputPane.tsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+  onSubmit,
+  notebookFile,
+}: StringInputPanelProps): JSX.Element => {
+  const [stringInput, setStringInput] = useState<string>(defaultInput);
+  const [formErrors, setFormErrors] = useState<string>("");
+  const [formErrorsDetails, setFormErrorsDetails] = useState<string>("");
+  const [isExecuting, setIsExecuting] = useState<boolean>(false);
+
+  const submit = () => {
+    console.log("stringInput", stringInput);
```
Comment (southpolesteve) [inline src/Explorer/Panes/StringInputPane/StringInputPane.tsx:44]: Remove extra console.log

File: src/Explorer/Panes/StringInputPane/StringInputPane.tsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+  defaultInput,
+  onSubmit,
+  notebookFile,
+}: StringInputPanelProps): JSX.Element => {
+  const [stringInput, setStringInput] = useState<string>(defaultInput);
+  const [formErrors, setFormErrors] = useState<string>("");
+  const [formErrorsDetails, setFormErrorsDetails] = useState<string>("");
+  const [isExecuting, setIsExecuting] = useState<boolean>(false);
+
+  const submit = () => {
```
Comment (southpolesteve) [inline src/Explorer/Panes/StringInputPane/StringInputPane.tsx:43]: Maybe this an async function

File: src/Explorer/Panes/StringInputPane/StringInputPane.tsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+      return;
+    } else {
+      setFormErrors("");
+      setFormErrorsDetails("");
+    }
+
+    const clearMessage = logConsoleProgress(`${inProgressMessage} ${stringInput}`);
+    setIsExecuting(true);
+    onSubmit(notebookFile, stringInput)
+      .then((value: NotebookContentItem) => {
```
Comment (southpolesteve) [inline src/Explorer/Panes/StringInputPane/StringInputPane.tsx:58]: Use async/await

**Review by southpolesteve — CHANGES_REQUESTED** [review]

File: src/Explorer/Panes/StringInputPane/StringInputPane.tsx (hunk 1 above)
Comment (hardiknai-techm) [reply src/Explorer/Panes/StringInputPane/StringInputPane.tsx:44]: done

File: src/Explorer/Panes/StringInputPane/StringInputPane.tsx (hunk 2 above)
Comment (hardiknai-techm) [reply src/Explorer/Panes/StringInputPane/StringInputPane.tsx:43]: yes, i have make async function

File: src/Explorer/Panes/StringInputPane/StringInputPane.tsx (hunk 3 above)
Comment (hardiknai-techm) [reply src/Explorer/Panes/StringInputPane/StringInputPane.tsx:58]: done

**Pull request merged**